source/Ship.cpp

```diff
@@ -80,19 +80,44 @@ void Ship::Load(const DataNode &node)
 	bool hasDescription = false;
 	for(const DataNode &child : node)
 	{
-		const string &key = child.Token(0);
+		bool add = (child.Token(0) == "add");
+		if(add && child.Size() < 2)
+		{
+			child.PrintTrace("Skipping " + child.Token(0) + " with no key given:");
+			continue;
+		}
+		// Get the key and value (if any).
+		const string &key = child.Token(add ? 1 : 0);
+		int valueIndex = add ? 2 : 1;
+		bool hasValue = (child.Size() > valueIndex);
+		const string &value = child.Token(hasValue ? valueIndex : 0);
```

Comment:
Given that attributes are the only thing you can add, you could just leave `key` as it is and avoid all the changes below. (The attributes line would become `if(key == "attributes" || add)`.)